### PR TITLE
Delete only if primary keys exist

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement

### CODE REVIEW

File: internal/core/src/segcore/segment_c.h [hunk 1]
```diff
@@ -120,6 +120,12 @@ AddFieldDataInfoForSealed(CSegmentInterface c_segment,
                           CLoadFieldDataInfo c_load_field_data_info);
 
 //////////////////////////////    interfaces for SegmentInterface    //////////////////////////////
+CStatus
+ExistPk(CSegmentInterface c_segment,
+        const uint8_t* raw_ids,
+        const uint64_t size,
+        bool* results);
```
Comment (jiaoew1991) [inline internal/core/src/segcore/segment_c.h:127]: bool* ?

Comment (yah01) [reply internal/core/src/segcore/segment_c.h:127]: yes it queries for an ID batch, the `bool* results` points to a bool array

**Review by jiaoew1991 — APPROVED** [review]
/lgtm

**Pull request merged**